services/cd-service/pkg/service/batch.go

```diff
@@ -139,6 +139,14 @@ func (d *BatchServer) processAction(
 		return &repository.CreateUndeployApplicationVersion{
 			Application: act.Application,
 		}, nil
+	case *api.BatchAction_Undeploy:
+		act := action.Undeploy
+		if err := ValidatePrepareUndeploy(act.Application); err != nil {
```

Comment:
```suggestion
		if err := ValidatePrepareUndeploy(act.Application); err != nil {
```
Either you rename this to `ValidateApplication`, or you make a seperate function that is called `ValidateUndeploy`.
As it is now, this is a bit misleading.